### PR TITLE
Move `HOMEBREW_NO_INSTALL_FROM_API` defaults to shell

### PR DESCRIPTION
**Pull request by Bo98**
This is a bit more involved than I thought it would be because we need to now have default prefix & supported OS information in shell too.

Though we do need to move the defaults to shell or `brew update` and `formula_path.sh` won't operate correctly - we've left those users in a "frozen state" where they are no longer receiving any formula or cask updates.

### CODE REVIEW

File: Library/Homebrew/env_config.rb [hunk 1]
```diff
@@ -502,9 +502,6 @@ def automatically_set_no_install_from_api?
 
     sig { returns(T::Boolean) }
     def install_from_api?
-      return false if OS.unsupported_configuration?
-      return false unless Homebrew.default_prefix?
-
       ENV["HOMEBREW_NO_INSTALL_FROM_API"].blank?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/env_config.rb:505]: Given this we may want to consider replacing all `install_from_api?` calls to `!no_install_from_api?` and deleting this

File: Library/Homebrew/os/mac/version.rb [hunk 2]
```diff
@@ -17,16 +17,16 @@ class Version < ::Version
       # and also update references in docs/Installation.md,
       # https://github.com/Homebrew/install/blob/HEAD/install.sh and
       # MacOSVersions::SYMBOLS
```
Comment (MikeMcQuaid) [inline Library/Homebrew/os/mac/version.rb:19]: Above comment needs updated

File: Library/Homebrew/os/mac/version.rb [hunk 3]
```diff
@@ -17,16 +17,16 @@ class Version < ::Version
...
-      NEWEST_UNSUPPORTED = "14"
+      NEWEST_UNSUPPORTED = ENV.fetch("HOMEBREW_MACOS_NEWEST_UNSUPPORTED").freeze
       private_constant :NEWEST_UNSUPPORTED
 
       # TODO: bump version when new macOS is released and also update
       # references in docs/Installation.md and
       # https://github.com/Homebrew/install/blob/HEAD/install.sh
```
Comment (MikeMcQuaid) [inline Library/Homebrew/os/mac/version.rb:25]: Above comment needs updated

File: Library/Homebrew/os/mac/version.rb [hunk 4]
```diff
@@ -17,16 +17,16 @@ class Version < ::Version
...
-      OLDEST_SUPPORTED = "11"
+      OLDEST_SUPPORTED = ENV.fetch("HOMEBREW_MACOS_OLDEST_SUPPORTED").freeze
       private_constant :OLDEST_SUPPORTED
 
-      OLDEST_ALLOWED = "10.11"
+      OLDEST_ALLOWED = ENV.fetch("HOMEBREW_MACOS_OLDEST_ALLOWED").freeze
```
Comment (MikeMcQuaid) [inline Library/Homebrew/os/mac/version.rb:29]: Think these are probably worth moving to `global.rb` after that so we have all these required `ENV.fetch` in a single place.

**Review by MikeMcQuaid — COMMENTED** [review]
Nice work so far @Bo98, lovely cleanup!

File: Library/Homebrew/os/mac/version.rb (hunk 4 above)
Comment (Bo98) [reply Library/Homebrew/os/mac/version.rb:29]: Yeah I initially thought these were macOS only but apparently there's a code path where Linux gets here. global would make it obvious to set the envs on all OS.

**Pull request merged**